src/agent/src/policy.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2023 Microsoft Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::{anyhow, Result};
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use tokio::time::{sleep, Duration};
+
+#[cfg(feature = "agent-policy-debug")]
+use tokio::io::AsyncWriteExt;
+
+static EMPTY_JSON_INPUT: &str = "{\"input\":{}}";
+
+static OPA_DATA_PATH: &str = "/data";
+static OPA_POLICIES_PATH: &str = "/policies";
+
+/// Convenience macro to obtain the scope logger
+macro_rules! sl {
+    () => {
+        slog_scope::logger()
+    };
+}
+
+/// Example of HTTP response from OPA: {"result":true}
+#[derive(Debug, Serialize, Deserialize)]
+struct AllowResponse {
+    result: bool,
+}
+
+/// Singleton policy object.
+#[derive(Debug)]
+pub struct AgentPolicy {
+    /// When true policy errors are ignored, for debug purposes.
+    allow_failures: bool,
+
+    /// OPA path used to query if an Agent gRPC request should be allowed.
+    /// The request name (e.g., CreateContainerRequest) must be added to
+    /// this path.
+    query_path: String,
+
+    /// OPA path used to add or delete a rego format Policy.
+    policy_path: String,
+
+    /// Client used to connect a single time to the OPA service and reused
+    /// for all the future communication with OPA.
+    opa_client: Client,
+
+    /// "/tmp/policy.txt" log file for policy activity.
+    #[cfg(feature = "agent-policy-debug")]
+    log_file: Option<tokio::fs::File>,
+}
+
+impl AgentPolicy {
+    /// Create AgentPolicy object.
+    pub fn new(opa_uri: &str, coco_policy: &str) -> Result<Self> {
+        Ok(AgentPolicy {
+            allow_failures: false,
+            query_path: opa_uri.to_string() + OPA_DATA_PATH + coco_policy + "/",
+            policy_path: opa_uri.to_string() + OPA_POLICIES_PATH + coco_policy,
+            opa_client: Client::builder().http1_only().build().unwrap(),
+
+            #[cfg(feature = "agent-policy-debug")]
+            log_file: None,
+        })
+    }
+
+    /// Wait for OPA to start and connect to it.
+    pub async fn initialize(&mut self) -> Result<()> {
+        #[cfg(feature = "agent-policy-debug")]
+        {
+            self.log_file = Some(tokio::fs::OpenOptions::new()
+                .write(true)
+                .truncate(true)
+                .create(true)
+                .open("/tmp/policy.txt")
+                .await
+                .unwrap());
+        }
+
+        let policy = tokio::fs::read_to_string("/etc/kata-opa/default-policy.rego")
```

Comment:
Currently we are using a lot of configurations/tempfiles/... in CoCo. It would be better that we put all these files under a same parent dir for better maintance, like `/run/confidential-containers/policy.txt` from https://github.com/kata-containers/kata-containers/pull/7570/files#diff-803c177f300cad05867e555f0b0080ee238df2ef218984786620f35d2198af33R10

However, if the policy file is assumed to be placed inside the guest image, `/etc` is a good choice.

Reply:
In the current design, an agent built using AGENT_POLICY=yes always starts with a default Policy that is baked in the Guest image. A typical default Policy is: https://github.com/microsoft/kata-containers/blob/cc-msft-prototypes/src/agent/samples/policy/set-policy-allowed/set-policy-allowed.rego

So, I think /etc is reasonable here. But, I wouldn't mind moving this file if other folks prefer it in a different location.